server/src/main/java/org/elasticsearch/common/geo/TriangleTreeReader.java

```diff
@@ -311,22 +314,23 @@ private boolean intersectsLine(int aX, int aY, int bX, int bY) {
          * Checks if the rectangle intersects the provided triangle
          **/
         private GeoRelation relateTriangle(int aX, int aY, boolean ab, int bX, int bY, boolean bc, int cX, int cY, boolean ca) {
-            // 1. query contains any triangle points
-            if (contains(aX, aY) || contains(bX, bY) || contains(cX, cY)) {
-                return GeoRelation.QUERY_CROSSES;
-            }
-
             // compute bounding box of triangle
             int tMinX = StrictMath.min(StrictMath.min(aX, bX), cX);
             int tMaxX = StrictMath.max(StrictMath.max(aX, bX), cX);
             int tMinY = StrictMath.min(StrictMath.min(aY, bY), cY);
             int tMaxY = StrictMath.max(StrictMath.max(aY, bY), cY);
 
-            // 2. check bounding boxes are disjoint
-            if (tMaxX < minX || tMinX > maxX || tMinY > maxY || tMaxY < minY) {
+
+            // 1. check bounding boxes are disjoint, where north and east boundaries are not considered as crossing
+            if (tMaxX <= minX || tMinX > maxX || tMinY > maxY || tMaxY <= minY) {
```

Comment:
We need to do a similar change for Tile2D#contains and Tile2D#relateLine

Reply:
I have deferred some of these changes as it isn't clear as to which behavior is preferred as previous expected tests are failing